python_modules/libraries/dagster-dbt/dagster_dbt/dagster_dbt_translator.py

```diff
@@ -288,6 +289,38 @@ def get_group_name(self, dbt_resource_props: Mapping[str, Any]) -> Optional[str]
         """
         return default_group_from_dbt_resource_props(dbt_resource_props)
 
+    @public
+    def get_code_version(self, dbt_resource_props: Mapping[str, Any]) -> Optional[str]:
+        """A function that takes a dictionary representing properties of a dbt resource, and
+        returns the Dagster code version for that resource.
+
+        Note that a dbt resource is unrelated to Dagster's resource concept, and simply represents
+        a model, seed, snapshot or source in a given dbt project. You can learn more about dbt
+        resources and the properties available in this dictionary here:
+        https://docs.getdbt.com/reference/artifacts/manifest-json#resource-details
+
+        This method can be overridden to provide a custom code version for a dbt resource.
+
+        Args:
+            dbt_resource_props (Mapping[str, Any]): A dictionary representing the dbt resource.
+
+        Returns:
+            Optional[str]: A Dagster code version.
+
+        Examples:
+            .. code-block:: python
+
+                from typing import Any, Mapping
+
+                from dagster_dbt import DagsterDbtTranslator
+
+
+                class CustomDagsterDbtTranslator(DagsterDbtTranslator):
+                    def get_code_version(self, dbt_resource_props: Mapping[str, Any]) -> Optional[str]:
+                        return "custom_code_version"
```

Comment:
It would be great to have an example that fits a real use case. I don't think this would ever be a static string.

Reply:
No problem! I've changed example from static string to overriding code_version with dbt node checksum. Is it better example?